src/test/java/com/clouway/chita/HttpClientTest.java

```diff
@@ -378,17 +384,20 @@ public void cacheReadingOfResponse() throws Exception {
   }
 
   @Test
-  public void readingRequestTimeout() throws Exception {
+  public void readingRequestTimeout() {
     HttpRequest post = httpRequest(new TargetUrl(serverUrl, serviceUrl))
             .post("")
             .readTimeout(1)
             .build();
 
     server.sleepFor(2000);
-    HttpResponse response = httpClient.execute(post);
-
-    assertThat(response.code(), is(408));
-    assertThat(response.isTimedOut(), is(true));
+    HttpResponse response = null;
+    try {
+      response = httpClient.execute(post);
+      fail("IOException was not thrown when connection timeouts ? ");
+    } catch (IOException e) {
+      assertThat(response, is(nullValue()));
```

Comment:
For example when we check if stb devices are online or not, the ACS responses with 408 if the device is offline. Currently we are not handling it this way but I think that we should, because now we return that the devices are offline even when there is no connection to the ACS which some times confuses the support guys that devices are offline when they are actually online but the ACS i just not responding.